### PR TITLE
Led.RGB: Add intensity method

### LINKED ISSUE #737: Add Led.RGB.intensity()
Similar to current `Led.brightness()` but has range 0-100 (percent).

Should consider deprecating `.brightness()` completely and add `.intensity(0-100)` to Led as well.

Reasoning: "brightness" can be misinterpreted. "intensity" is standard term in lighting industry.

### PR DESCRIPTION
**Pull request by scottgonzalez**
Fixes #737 

Alternative to #818, storing the intensity as separate state from color.

### CODE REVIEW

**Comment by Resseguie** [comment]
Captures my version of "intensity" well.

Only discussion point for me is the behavior when you call `led.off()` then 'led.intensity(50)`, is it still "off" and waiting to be turned "on" and shown at 50%, or are `.off()` and `.on()` really shorthand for `.intensity(0)` and `.intensity(100)` respectively? (And thus, the led would immediately be lit at 50% when `.intensity(50)` is called, regardless of on/off state.)

**Comment by scottgonzalez** [comment]
@rwaldron Did you get a chance to look at this?

**Comment by dtex** [comment]
@Resseguie Motor has a similar thing in that a motor will remember it's speed when stopped so that if started later (with no params) it will resume at the same speed as before. I think you need to store the intensity independently of on/off state.

**Comment by Resseguie** [comment]
@dtex good point, thanks!

File: lib/led/rgb.js [hunk 1]
```diff
@@ -405,4 +419,18 @@ RGB.prototype.stop = function() {
...
 };
 
+RGB.prototype.intensity = function(intensity) {
+  var state = priv.get(this);
+
+  if (arguments.length === 0) {
+    return state.intensity;
+  }
+
+  state.intensity = Math.min(100, Math.max(0, intensity));
```
Comment (dtex) [inline lib/led/rgb.js]: This pattern is also defined in fn.js. https://github.com/rwaldron/johnny-five/blob/master/lib/fn.js#L48-L50 Maybe we should call that for consistency and readability?

Comment (scottgonzalez) [inline lib/led/rgb.js]: done

**Comment by Resseguie** [comment]
@mize note it's ready for whitepoint now if you're still interested in adding that.

**Comment by Resseguie** [comment]
#847 is an example demonstrating use of `intensity`

**Comment by mize** [comment]
@Resseguie thanks, will do!

**Pull request merged**